From 338a08bffe970129f961ff574d86c205763554a7 Mon Sep 17 00:00:00 2001
From: Gu, Wangyi <wangyi.gu@intel.com>
Date: Wed, 21 Aug 2013 13:20:58 +0800
Subject: fix the TOMBSTONE issue due to file not exist

BZ: 125857

root cause:
if the file is not exist, onPrepareAsyncEvent will be failed, and video
source is not preared, so TOMBSTONE will happen in readFirstVideoFrame.
reproduce steps:
1. Create a new project from video editor
2. Choose a video clip from gallery to import. After tap a video clip from
gallery.
3. Use "rm" command to remove the clip from adb shell before all thumbnails
decode accomplished.
4. Tap play button

Category:aosp improvement
Domain: Video.Media-video editor
Origin: internal
Upstream-Candidate: Yes

Change-Id: I998d3382701078e8f7e0d6afff79dbe1a4c1b184
Orig-Change-Id: I58e6f36157b785d09da719c688c88caf1b14d3dd
Signed-off-by: Gu, Wangyi <wangyi.gu@intel.com>
Reviewed-on: http://android.intel.com:8080/126628
Reviewed-by: Feng, Wei <wei.feng@intel.com>
Reviewed-by: Shi, PingX <pingx.shi@intel.com>
Tested-by: Shi, PingX <pingx.shi@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 libvideoeditor/lvpp/PreviewPlayer.cpp |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/libvideoeditor/lvpp/PreviewPlayer.cpp b/libvideoeditor/lvpp/PreviewPlayer.cpp
index bc25090..7bc0f44 100755
--- a/libvideoeditor/lvpp/PreviewPlayer.cpp
+++ b/libvideoeditor/lvpp/PreviewPlayer.cpp
@@ -1466,6 +1466,11 @@ status_t PreviewPlayer::setImageClipProperties(uint32_t width,uint32_t height) {
 status_t PreviewPlayer::readFirstVideoFrame() {
     ALOGV("readFirstVideoFrame");
 
+    if (!(mFlags & PREPARED)) {
+        ALOGE("The player is not prepared");
+        return UNKNOWN_ERROR;
+    }
+
     if (!mVideoBuffer) {
         MediaSource::ReadOptions options;
         if (mSeeking != NO_SEEK) {
-- 
1.7.4.1

